messages/src/lib.rs

```diff
@@ -39,7 +39,7 @@ pub mod taskprov;
 mod tests;
 
 /// Messages which have an HTTP media type associated with them.
-pub trait MediaType: Encode {
+pub trait MediaType {
```

Comment:
We don't actually need `MediaType` to depend on `Encode`, and in fact by removing this bound it's easier to use `EncodeKeyConfig` in `CachedResource`.